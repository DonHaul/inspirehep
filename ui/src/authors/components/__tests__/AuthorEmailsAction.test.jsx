import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';

import AuthorEmailsAction from '../AuthorEmailsAction';

describe('AuthorEmailsAction', () => {
  it('renders multiple current emails in a dropdown', () => {
    const emails = fromJS([
      { value: 'dude@email.cern' },
      { value: 'other-dude@email.cern' },
    ]);
    const { asFragment } = render(<AuthorEmailsAction emails={emails} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders single email', () => {
    const emails = fromJS([{ value: 'dude@email.cern' }]);
    const { asFragment } = render(<AuthorEmailsAction emails={emails} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
