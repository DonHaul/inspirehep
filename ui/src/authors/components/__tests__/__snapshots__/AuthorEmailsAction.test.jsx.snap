// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorEmailsAction renders multiple current emails in a dropdown 1`] = `
<DocumentFragment>
  <span
    class="__UserAction__"
  >
    <button
      class="ant-btn ant-btn-default ant-dropdown-trigger"
      type="button"
    >
      <span
        aria-label="mail"
        class="anticon anticon-mail"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="mail"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
          />
        </svg>
      </span>
    </button>
  </span>
</DocumentFragment>
`;

exports[`AuthorEmailsAction renders single email 1`] = `
<DocumentFragment>
  <span
    class="__UserAction__"
  >
    <a
      href="mailto:dude@email.cern"
      target="_blank"
    >
      <span
        aria-label="mail"
        class="anticon anticon-mail"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="mail"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
          />
        </svg>
      </span>
    </a>
  </span>
</DocumentFragment>
`;
